### PR TITLE
Rename local variables and clean up some code

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Remove local variable as mentioned #763

### CODE REVIEW

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java [hunk 1]
```diff
@@ -130,10 +130,9 @@ protected boolean commonAssignmentCheck(
         if (TreeUtils.isFieldAccess(varTree)) {
             // cast is safe: a field access can only be an IdentifierTree or MemberSelectTree
             ExpressionTree lhs = (ExpressionTree) varTree;
-            ExpressionTree y = valueExp;
             VariableElement el = TreeUtils.variableElementFromUse(lhs);
             AnnotatedTypeMirror xType = atypeFactory.getReceiverType(lhs);
-            AnnotatedTypeMirror yType = atypeFactory.getAnnotatedType(y);
+            AnnotatedTypeMirror yType = atypeFactory.getAnnotatedType(valueExp);
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java:135]: The name `yType` is now a bit unmotivated.
Can you rename both `xType` and `yType` to make this more meaningful? Maybe `lhsType` and `valueExpType`? Other ideas?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks! Only one comment.

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java (hunk 1 above)
Comment (Ao-senXiong) [reply checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java:135]: @wmdietl The name make sense to me. However, I am wondering why in this method the argument is `valueExp` and in super method it is `valueExpTree`.

Comment (wmdietl) [reply checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java]: Sure, make that consistent too.

Comment (Ao-senXiong) [reply checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java]: Done.

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java [hunk 2]
```diff
@@ -123,29 +123,28 @@ protected void checkThisOrSuperConstructorCall(
...
             Object... extraArgs) {
         // field write of the form x.f = y
         if (TreeUtils.isFieldAccess(varTree)) {
             // cast is safe: a field access can only be an IdentifierTree or MemberSelectTree
             ExpressionTree lhs = (ExpressionTree) varTree;
-            ExpressionTree y = valueExp;
             VariableElement el = TreeUtils.variableElementFromUse(lhs);
-            AnnotatedTypeMirror xType = atypeFactory.getReceiverType(lhs);
-            AnnotatedTypeMirror yType = atypeFactory.getAnnotatedType(y);
+            AnnotatedTypeMirror lhsType = atypeFactory.getReceiverType(lhs);
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java:134]: This uses `getReceiver`, so it's not the type of the `lhs`, is it? It's the receiver of the LHS?

Comment (Ao-senXiong) [reply checker/src/main/java/org/checkerframework/checker/initialization/InitializationVisitor.java]: okay, renamed to `lhsReceiverType`.

File: javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java [hunk 3]
```diff
@@ -148,9 +148,7 @@ private TreeUtils() {
             TREEMAKER_SELECT =
                     TreeMaker.class.getMethod("Select", JCExpression.class, Symbol.class);
         } catch (NoSuchMethodException e) {
-            Error err = new AssertionError("Unexpected error in TreeUtils static initializer");
-            err.initCause(e);
-            throw err;
+            throw (Error) new AssertionError("Unexpected error in TreeUtils static initializer", e);
```
Comment (wmdietl) [inline javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java:151]:
```suggestion
            throw new AssertionError("Unexpected error in TreeUtils static initializer", e);
```

That upcast seems unnecessary. Is your IDE not complaining about that?

Comment (Ao-senXiong) [reply javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java:151]: Just checked it did not complain about the cast.

Comment (wmdietl) [reply javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java:151]: Do you agree with removing the cast? Or do you see some benefit in keeping it?

Comment (Ao-senXiong) [reply javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java:151]: Sure, remove it is better for clean code.

**Review by wmdietl — APPROVED** [review]
Thanks! The new names are much better.

**Pull request merged**